backend/plan/urls.py

```diff
@@ -12,5 +17,7 @@
 urlpatterns = [
     path("", TemplateView.as_view(template_name="plan/build/index.html")),
     path("recommendations/", recommend_courses_view, name="recommend-courses"),
+    path("shared/get_schedule", get_shared_schedule, name="get-shared-schedule"),
+    path("shared/set_schedule", set_shared_schedule, name="set-shared-schedule"),
```

Comment:
Do we need a route for this? Couldn't we just use the existing update schedule route to directly set `is_shared = True`? If so, we might want to add some validation to [that route](https://github.com/pennlabs/penn-courses/blob/494cb5326617a02363afa9267b064de7f6d16612/backend/plan/views.py#L385) (create + update schedule) to return a better error message if the at most one shared schedule constraint is violated (rather than bubbling up a 500 SQL exception), and document that case causing a 400 [here](https://github.com/pennlabs/penn-courses/blob/494cb5326617a02363afa9267b064de7f6d16612/backend/plan/views.py#L267), just like the semester check. Don't think we would need to even change the [serializer](https://github.com/pennlabs/penn-courses/blob/494cb5326617a02363afa9267b064de7f6d16612/backend/plan/serializers.py), as all fields are included by default.

Reply:
Should we keep the get-shared-schedule route to ensure that we can get the shared schedule? I do agree that we can handle the set-shared-schedule tasks in the update route for the class view of Schedule.

Reply:
Yeah I think we do need a separate route for getting shared schedules, bc the GET method on the schedules viewset will return the user's schedules, not schedules that are shared with the user.